lib/schema/facter.yaml

```diff
@@ -184,6 +184,15 @@ chassistype:
     caveats: |
         Linux: kernel 2.6+ is required due to the reliance on sysfs.
 
+cloud:
+    type: map
+    description: Information about the cloud instance of the node. This is currently only populated on linux nodes running in Microsoft Azure.
+    validate: false
```

Comment:
We only set `validate` to false on facts where we don't know what the names of their elements will be (ec2 metadata is filled from the metadata end-point). In this case we know exactly what it'll be - `provider` - so we should include that in testing.